sklearn/neighbors/dist_metrics.pyx

```diff
@@ -1048,6 +1052,22 @@ cdef class HaversineDistance(DistanceMetric):
 #            norm2 += x2[j] * x2[j]
 #        return 1.0 - d / sqrt(norm1 * norm2)
 
+#------------------------------------------------------------
+# Arccos Distance
+#  D(x, y) = arccos(dot(x, y) / (|x| * |y|)) / PI
+# This is the metric version of cosine distance, i.e. the metric
+# most suitable when you wish to use cosine distance.
+
+cdef class ArccosDistance(DistanceMetric):
+    cdef inline DTYPE_t dist(self, DTYPE_t* x1, DTYPE_t* x2, ITYPE_t size) nogil except -1:
+        cdef DTYPE_t d = 0, norm1 = 0, norm2 = 0
+        cdef np.intp_t j
+        for j in range(size):
+            d += x1[j] * x2[j]
```

Comment:
Just a comment, if `size` is large enough, and assuming the data is normalized, the dot product could possibly be faster computed with the `ddot` from BLAS ( [imported in cython, for instance, here](https://github.com/scikit-learn/scikit-learn/blob/4d9a12d175a38f2bcb720389ad2213f71a3d7697/sklearn/linear_model/cd_fast.pyx#L116)), since `DTYPE_t == float64` as far as I understand.